restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/Conference.java

```diff
@@ -577,4 +600,59 @@ private void updateConferenceStatus(ConferenceStateChanged.State state){
             dao.updateConferenceDetailRecordStatus(cdr);
         }
     }
+
+    /**
+     * startConferenceTimer - conference should expire after 4 hours/(configurable)
+     */
+    private void startConferenceTimer() {
+        final long conferenceTotalLifeInMillis = RestcommConfiguration.getInstance().getMain().getConferenceTimeout()*1000;
+        final long conferenceRemainingLife =  conferenceTotalLifeInMillis - conferenceAge();
+        context().setReceiveTimeout(Duration.create(conferenceRemainingLife, TimeUnit.MILLISECONDS));
+        if(logger.isInfoEnabled())
+            logger.info(String.format("conference timer started for: %s milliseconds", conferenceRemainingLife));
+    }
+
+    /**
+     * @return conference age in milli seconds
+     */
+    private long conferenceAge() {
+        if(sid != null){
+            final ConferenceDetailRecordsDao dao = storage.getConferenceDetailRecordsDao();
+            ConferenceDetailRecord cdr = dao.getConferenceDetailRecord(sid);
+            if(cdr == null){
+                if(logger.isDebugEnabled())
+                    logger.debug(String.format("Conference cdr is null for sid: %s", sid));
+            }else{
+                return DateTime.now().getMillis()-cdr.getDateCreated().getMillis();
+            }
+        }
+        return 0;
+    }
+
+    protected ActorRef conferenceApiClient() {
+        final Props props = new Props(new UntypedActorFactory() {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public UntypedActor create() throws Exception {
+                return new ConferenceApiClient(accountSid, friendlyName, sid, storage);
+            }
+        });
+        return getContext().actorOf(props);
+    }
+
+    @Override
+    public void postStop() {
+        if (!fsm.state().equals(uninitialized)) {
+            if(logger.isInfoEnabled()) {
+                logger.info("Conference: " + self().path()
+                    + "At the postStop() method.");
+            }
+            if(conferenceApiClient != null && !conferenceApiClient.isTerminated())
+                getContext().stop(conferenceApiClient);
```

Comment:
@maria-farooq instead of waiting for the Conference actor to get destroyed in order to destroy the conferenceApiClient, we can make use of the fact the ConferenceApiClient will send the DownloaderResponse to the Confernece actor and kill the actor at this point

Reply:
@gvagenas i have changed both `CallApiClient` and `ConferenceApiClient` to be disposable (used for one time request), they will clean them selves after sending back `DownloaderResponse`